src/applications/mhv/medical-records/reducers/allergies.js

```diff
@@ -58,6 +58,14 @@ export const allergyReducer = (state = initialState, action) => {
           }) || [],
       };
     }
+    case Actions.Allergies.CLEAR_DETAIL: {
+      return {
+        ...state,
+        allergyDetails: () => {
+          return undefined;
+        },
```

Comment:
You should be able to use `allergyDetails: undefined` here instead of returning a function.